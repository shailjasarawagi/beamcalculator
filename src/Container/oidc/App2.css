.App2 {
    text-align: center;
    padding-top:100px;
  }
  
  .App2-logo {
    animation: NewApp-logo-spin infinite 20s linear;
    height: 40vmin;
    pointer-events: none;
  }
  
  .App2-header {
    background-color: #282c34;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
  }
  
  .App2-link {
    color: #61dafb;
  }
  
  @keyframes NewApp-logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
  
