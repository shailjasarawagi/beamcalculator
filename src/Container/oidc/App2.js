/**
 * This manages the redirection of url when user is not authenticated
 */
import React, { Component } from 'react';
import classes from './App2.css';
//import axios from '../../hoc/axios-orders';
import * as functions from '../../Functions/Function';
import { withRouter } from 'react-router-dom';

/**send_request() is called when user is not authenticated and a new window is opened */
class App2 extends Component {

  componentDidMount() {
    this.status = "KO";
    this.pollInterval = null;
    this.polling_id = null;
    this.send_request();
  }

  state = {
    redirect_url: null
  }
  
  /**This opens a new window when user not authenticated */
  send_request = () => {
    // console.log("request sent")
    // axios.get('/oidc/auth/request/', {})
    //   .then(response => {
    //     // console.log("hello response", response.data);
    //     // this.setState({
    //     //   redirect_url: "df"
    //     // })
    //     // this.pooling_id = response.data.pooling_id;
    //     this.window = window.open(response.data.redirect_url, "_self");
    //     // localStorage.setItem("pollingId", this.pooling_id);
    //   })
    //   .catch((error) => {
    //     console.log("error", error);
    //   });

        // const url=process.env.REACT_APP_BACKEND_URL;
        const url=functions.getBackendDomain().baseURL;

      window.open(url+"/oidc/auth/request/", "_self");
  }

 
  render() {
    return (
      <div className={classes.App2}>
      </div>
    )
  }
}

export default withRouter(App2)