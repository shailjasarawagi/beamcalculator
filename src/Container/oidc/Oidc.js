/**
 * Used for Routing the pages of App with token validation of user and request validation 
 */
import { Route, Switch, Redirect } from "react-router-dom";
import React, { Component } from "react";
import * as functions from '../../Functions/Function';
import App2 from './App2';
import axios from "../../hoc/Axios-orders";
import Beam from '../BeamForm/beam';

class Oidc extends Component {

  state = {
    display: null
  }

  /**Defines routing(switch) component in mount */
  componentDidMount() {
    let str = window.location.pathname;
    let token = str.split("/")
    const url = "/auth/token-verify";
    let display1 = (<Switch>

      <Route exact path="/" component={() => <App2 />} />
      <Redirect to="/" />
    </Switch>);
    let displayer = (<Switch>
      <Route exact path="/beam" component={Beam} />
      <Route exact path="/" component={Beam} />
      <Redirect to="/beam" />
    </Switch>);

    /**token[2] contains token  which expires after 10 min */
    if (token[2])  //check if token exists in url
    {

      localStorage.setItem("token", token[2])
      let header1 = functions.getToken();
      // console.log("retrieved token using getToken", header1)
      axios.get(url,
        { headers: header1 }
      ).then(
        (response) => {
          localStorage.setItem("exp_time", response.data.exp_time * 1000);
          setTimeout(this.refresh_token, response.data.exp_time * 1000 - (new Date().getTime() + 20000));
          // setTimeout(this.refresh_token, response.data.exp_time * 1000 - (new Date().getTime() + 540000))
          this.setState({ display: displayer })
        }, (error1) => {
          if (error1.response.statusText.trim() === "Forbidden".trim()) {
            this.setState({ display: display1 })
          }//if token is expired and forbidden, direct to login page(App2)
        }
      )
    }
    else //if url does not contain token from backend
    {
      this.setState({ display: display1 })
    }
  }


  render() {
    return <div>{this.state.display}</div>;
  }

  /**again token is refeshed after expire */
  refresh_token = () => {
    let refresh_url = "/oidc/refresh-token/";
    let header2 = functions.getToken();
    // console.log("currently valid token", header2)
    axios.get(refresh_url, { headers: header2 }).then((response) => {
      // let encoded_token = window.btoa(response.data.access_token)
      let encoded_token2 = response.data.access_token
      // console.log("token for refresh", encoded_token2)
      localStorage.clear("token");
      localStorage.setItem("token", encoded_token2);
      localStorage.setItem("exp_time", response.data.exp_time)
      // setTimeout(this.refresh_token, response.data.exp_time * 1000 - (new Date().getTime() + 540000));
      setTimeout(this.refresh_token, response.data.exp_time * 1000 - (new Date().getTime() + 20000));


    })
  }


}

export default Oidc;