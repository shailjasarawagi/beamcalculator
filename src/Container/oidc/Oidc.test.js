import React from "react";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Oidc from "../oidc/Oidc";
// import App from "../src/App";
import { MemoryRouter } from 'react-router';
import PipingDB from "../PipingDB/PipingDB";
import App2 from "../oidc/App2";

configure({ adapter: new Adapter() });

describe("<Oidc/>", () => {

    it('renders correctly', () => {
        const component = shallow(<Oidc />);
        expect(component).toMatchSnapshot();
    });


    // test('invalid path should redirect to 404', () => {
    //     const wrapper = mount(
    //         <MemoryRouter initialEntries={['/']}>
    //             <Oidc />
    //         </MemoryRouter>
    //     );
    //     expect(wrapper.find(PipingDB)).toHaveLength(0);
    //     expect(wrapper.find(App2)).toHaveLength(1);
    // });

    // test('valid path should not redirect to 404', () => {
    //     const wrapper = mount(
    //         <MemoryRouter initialEntries={['/pdb']}>
    //             <Oidc />
    //         </MemoryRouter>
    //     );
    //     expect(wrapper.find(PipingDB)).toHaveLength(1);
    //     expect(wrapper.find(App2)).toHaveLength(0);
    // });

});