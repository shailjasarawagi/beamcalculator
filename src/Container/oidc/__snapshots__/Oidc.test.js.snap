// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Oidc/> renders correctly 1`] = `ShallowWrapper {}`;
